### PR TITLE
Allow INSERT into if type-only coercion possible to table column types

### PR DESCRIPTION
**Pull request by losipiuk**
Allow INSERT into if type-only coercion possible to table column types

This patch relaxes limitation that when INSERT INTO statement is called
values/query types must exactly match table column types.
Now we allow values/query types to be type-only coercible to table
column types.

This is intermediate step until all coercions are allowed.

### CODE REVIEW

**Comment by losipiuk** [comment]
@sopel39, @kokosing take a look

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java [hunk 1]
```diff
@@ -455,6 +458,23 @@ protected TupleDescriptor visitInsert(Insert insert, AnalysisContext context)
         return new TupleDescriptor(Field.newUnqualified("rows", BIGINT));
     }
 
+    private boolean typesMatchForInsert(Iterable<Type> tableTypes, Iterable<Type> queryTypes)
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java]: can you extract this method as more generic one to match invocation

```
elementsEqual(tableTypes, queryTypes, (o1, o2) -> o1.equals(o2) && !FunctionRegistry.isTypeOnlyCoercion(o1,o2));
```

**Comment by kokosing** [comment]
lgtm
Does it work? I mean do problematic insert statements pass?

**Comment by sopel39** [comment]
lgtm,
the same issue is in: presto-hive/src/main/java/com/facebook/presto/hive/HivePageSink.java (see my patch). However we don't have access to FunctionRegistry there.

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java (hunk 1 above)
Comment (losipiuk) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java]: Not doing that.
naming method elementsEqual is misleading. As "equality" function we are passing to it is not symmetric.

**Pull request closed without merging**